src/main/java/net/aufdemrand/denizen/events/scriptevents/BucketFillScriptEvent.java

```diff
@@ -0,0 +1,115 @@
+package net.aufdemrand.denizen.events.scriptevents;
+
+import net.aufdemrand.denizen.BukkitScriptEntryData;
+import net.aufdemrand.denizen.objects.*;
+import net.aufdemrand.denizen.utilities.DenizenAPI;
+import net.aufdemrand.denizencore.events.ScriptEvent;
+import net.aufdemrand.denizencore.objects.dObject;
+import net.aufdemrand.denizencore.scripts.ScriptEntryData;
+import net.aufdemrand.denizencore.scripts.containers.ScriptContainer;
+import net.aufdemrand.denizencore.utilities.CoreUtilities;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerBucketFillEvent;
+
+import java.util.HashMap;
+
+public class BucketFillScriptEvent extends ScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player fills bucket
+    // player fills bucket with <material>
+    //
+    // @Triggers when a player fills a bucket.
+    //
+    // @Cancellable true
+    //
+    // @Context
+    // <context.item> returns the dItem of the bucket.
+    // <context.location> returns the dLocation of the block clicked with the bucket.
+    // <context.relative> returns the dLocation of the block in front of the clicked block.
+    // <context.material> returns the dMaterial the the bucket is being filled with.
+    //
+    // -->
+
+
+    public BucketFillScriptEvent() {
+        instance = this;
+    }
+
+    public static BucketFillScriptEvent instance;
+    public dEntity entity;
+    public dItem item;
+    public dMaterial material;
+    public dLocation location;
+    public dLocation relative;
+    public PlayerBucketFillEvent pEvent;
```

Comment:
just call it `event`, I only used `pEvent` that one time because there were two different events.